Anime-App/Feature/Anime/View/AnimeViewController.swift

```diff
@@ -0,0 +1,144 @@
+//
+//  ViewController.swift
+//  Anime-App
+//
+//  Created by Tobi Adegoroye on 09/08/2023.
+//
+
+import UIKit
+
+class AnimeViewController: UIViewController {
+    
+    var searchTask: DispatchWorkItem?
+    private let animeViewModel = AnimeViewModel()
+    private var animes: [Anime] = [] {
+        didSet {
+            DispatchQueue.main.async {
+                self.tableView.reloadData()
+            }
+        }
+    }
+    
+    private var tableView: UITableView = {
+        let tableView = UITableView()
+        tableView.translatesAutoresizingMaskIntoConstraints = false
+        tableView.register(AnimeTableViewCell.self, forCellReuseIdentifier: AnimeTableViewCell.cellID)
+        return tableView
+    }()
+ 
+    private let searchVc = UISearchController(searchResultsController: nil)
+ 
+    override func viewDidLoad() {
+        super.viewDidLoad()
+        setup()
+    }
+}
+
+extension AnimeViewController: UISearchBarDelegate {
+    
+    func searchBar(_ searchBar: UISearchBar, textDidChange searchText: String) {
+        guard let searchText = searchBar.text?.lowercased(), !searchText.isEmpty else {
+            return
+        }
+        
+        guard let searchText = searchBar.text?.lowercased(), !searchText.isEmpty else {
+            return
+        }
+        self.searchTask?.cancel()
+
+        let task = DispatchWorkItem { [weak self] in
+            DispatchQueue.global(qos: .userInteractive).async { [weak self] in
+              //Use search text and perform the query
+        
+                self?.animeViewModel.searchAnime(query: searchText) { res in
+                    switch res {
+                    case .success(let anime):
+                        self?.animes = anime
+                        DispatchQueue.main.async {
```

Comment:
This is good but look into using didSet on variables since when the variable has been set you can get it to automatically reload the tableview here is a good tutorial, its a better way to refresh views since you may easily forgot to add this in but if you have it in a didset you can ensure the refresh will always fire

https://www.hackingwithswift.com/read/8/5/property-observers-didset